Makefile

```diff
@@ -5,20 +5,52 @@ BIN := build-conan/build/RelWithDebInfo/orbbec-module
 TAG_VERSION?=latest
 APPIMAGE := packaging/appimages/deploy/$(OUTPUT_NAME)-$(TAG_VERSION)-$(ARCH).AppImage
 ORBBEC_SDK_VERSION=v2.4.3
-ORBBEC_SDK_TIMESTAMP=202505191331
 ORBBEC_SDK_COMMIT=045a0e76
 ORBBEC_SDK_DIR=OrbbecSDK_$(ORBBEC_SDK_VERSION)_$(ORBBEC_SDK_TIMESTAMP)_$(ORBBEC_SDK_COMMIT)_$(OS)_$(ARCH)
 
+ifeq ($(OS),darwin)
+  ORBBEC_SDK_TIMESTAMP = 202505192200
+  ORBBEC_SDK_DIR = OrbbecSDK_$(ORBBEC_SDK_VERSION)_$(ORBBEC_SDK_TIMESTAMP)_$(ORBBEC_SDK_COMMIT)_macOS_beta
+else ifeq ($(OS),linux)
+  ifeq ($(ARCH),x86_64)
+    ORBBEC_SDK_TIMESTAMP = 202505191331
+    ORBBEC_SDK_DIR = OrbbecSDK_$(ORBBEC_SDK_VERSION)_$(ORBBEC_SDK_TIMESTAMP)_$(ORBBEC_SDK_COMMIT)_linux_$(ARCH)
+  else
+    $(error Unsupported architecture: $(ARCH))
+  endif
+else
+  $(error Unsupported OS: $(OS))
+endif
+
+
 .PHONY: build lint setup appimage
 
-module.tar.gz: $(APPIMAGE) meta.json
-	cp $(APPIMAGE) $(OUTPUT_NAME).AppImage
-	tar -czvf module.tar.gz $(OUTPUT_NAME).AppImage meta.json ./first_run.sh ./install_udev_rules.sh ./99-obsensor-libusb.rules
-	rm $(OUTPUT_NAME).AppImage
+module.tar.gz: $(BIN) meta.json
+ifeq ($(OS),linux)
+	cp $(APPIMAGE) $(OUTPUT_NAME)
+	tar -czvf module.tar.gz \
+		$(OUTPUT_NAME) \
+		meta.json \
+		./first_run.sh \
+		./install_udev_rules.sh \
+		./99-obsensor-libusb.rules
+	rm -f $(OUTPUT_NAME).AppImage
+else ifeq ($(OS),darwin)
+	install_name_tool -change $(ORBBEC_SDK_DIR)/lib/libOrbbecSDK.2.dylib @executable_path/lib/libOrbbecSDK.2.dylib $(BIN)
+	if ! otool -l $(BIN) | grep -A2 LC_RPATH | grep -q "@executable_path/lib"; then \
```

Comment:
why do we need this conditional? why not always add our rpath? my instincts tell me this may be more robust

Reply:
do you know when rpath is already added vs. when it's not?